pkg/proxy/proxy.go

```diff
@@ -561,17 +561,19 @@ func (p *Proxy) CreateOrUpdateRedirect(ctx context.Context, l4 policy.ProxyPolic
 
 	for nRetry := 0; nRetry < redirectCreationAttempts; nRetry++ {
 		if nRetry > 0 {
-			// an error occurred and we can retry
+			// an error occurred and we are retrying
 			scopedLog.WithError(err).Warningf("Unable to create %s proxy, retrying", ppName)
-			// Do not increment port for DNS when the port is set in config
-			if pp.proxyType != ProxyTypeDNS || option.Config.ToFQDNsProxyPort == 0 {
+		}
+		if !pp.configured {
```

Comment:
Follow up to my question in the PR discussion: 

Am i correct that this logic still increments the proxy port in case of an initial redirect creation if an Envoy startup error occurs in `createEnvoyRedirect` (e.g. memory pressure in case of embedded mode). In this case `pp.reservePort()` , which would set `pp.configured=true`, will not be called (only if no error occured).